addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -130,15 +130,13 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
-	clusterFSIDs := make(map[string]string)
-	klog.Infof("Fetching clusterFSIDs")
-	err = r.fetchClusterFSIDs(ctx, &mirrorPeer, clusterFSIDs)
-	if err != nil {
-		return ctrl.Result{Requeue: true}, fmt.Errorf("failed to get all cluster FSIDs, retrying again: %v", err)
-	}
-
-	klog.Info(clusterFSIDs)
 	if mirrorPeer.Spec.Type == multiclusterv1alpha1.Async {
+		clusterFSIDs := make(map[string]string)
+		klog.Infof("Fetching clusterFSIDs")
```

Comment:
I've moved it inside the check, this should fix the issue with the agent checking for clusterfsids

Reply:
This looks ok for now. I was trying to fetch the fsid for both async and sync setups using the cephcluster cr instead of the secrets but that can be done in a follow up patch.